ci/docker/debian-10-go-cgo.dockerfile

```diff
@@ -0,0 +1,34 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Why do we need this in addition to the existing `debian-10-go-cgo-python` dockerfile?

Reply:
The `debian-10-go-cgo-python` dockerfile installs pyarrow and the underlying go does not rely on linking against the libarrow.so objects.

This dockerfile installs libarrow directly via apt so that pkg-config can find libarrow.so since the allocator piece directly needs to link against libarrow.so to run which is explicitly different than the case for the `cgo-python` one which in which Go has no dependency on libarrow since it only needs the header files and .c/.cc files which are in the Go repo itself.

Reply:
Ok, but the comment is misleading then ("so we can use pyarrow").